testsuite/shared/src/main/java/org/jboss/as/test/integration/management/util/CustomCLIExecutor.java

```diff
@@ -255,34 +243,16 @@ private static String executeProcess(final CliCommandBuilder commandBuilder, fin
         boolean wait = true;
         int runningTime = 0;
         int exitCode = Integer.MIN_VALUE;
-        do {
-            try {
-                Thread.sleep(STATUS_CHECK_INTERVAL);
-            } catch (InterruptedException e) {
-            }
-            runningTime += STATUS_CHECK_INTERVAL;
-            try {
+        try {
+            boolean finished = cliProc.waitFor(CLI_PROC_TIMEOUT, TimeUnit.MILLISECONDS);
+            if (finished) {
                 exitCode = cliProc.exitValue();
-                wait = false;
-            } catch (IllegalThreadStateException e) {
-                // cli still working
-                if (failureResponse != null) {
-                    try {
-                        final OutputStream stdin = cliProc.getOutputStream();
-                        stdin.write(failureResponse.getBytes(StandardCharsets.UTF_8));
-                        stdin.write(NEW_LINE);
-                        stdin.flush();
-                    } catch (IOException e1) {
-                        LOGGER.debug(out.toString(), e);
-                    }
-                }
             }
-            if (runningTime >= CLI_PROC_TIMEOUT) {
-                cliProc.destroy();
-                wait = false;
-                LOGGER.info("A timeout has occurred while invoking CLI command.");
-            }
-        } while (wait);
+        } catch (InterruptedException e1) {
+            new RuntimeException("Interupted while waiting", e1);
```

Comment:
Even better it should probably use `Assert.fail()` since it's in the test suite.